docs/working-drafts/issue-29-erratum.md.adoc

```diff
@@ -0,0 +1,135 @@
+discussed below in Section (new sub-section of 9.5).
+
+A new sub-section is added: Evaluation Order for Statement Annotations
+for Elements with dfdl:lengthKind 'prefixed'.
+
+This new sub-section contains the content below.
+
+When an element or element reference has lengthKind 'prefixed' there
+can be additional statement annotations associated with the prefix
+length type.
+
+Definition: the resolved set of statement annotations for the simple
+type that is identified by the dfdl:prefixLengthKind property are
+called the _prefix statement annotations_.
+
+Definition: the resolve set of statement annotations for the nested
+prefix simple type that is identified by the dfdl:prefixLengthKind
+property on a simple type itself having lengthKind 'prefixed', are
+called the _prefix-prefix statement annotations_.
+
+For statement annotations associated with any prefix length, any DFDL
+expressions evaluated cannot refer to the DFDL Infoset using path
+expressions.
+
+The value of the prefix (or prefix-prefix) itself can be referenced
+using the expression ".".
+
+When parsing the value of the "." expression for a prefix (or
+prefix-prefix) quasi-element is as before any adjustment (subtraction)
+from the value based on dfdl:prefixIncludesPrefixLength.
+
+When unparsing the value of the "." expression for a prefix (or
+prefix-prefix) quasi-element is as after any adjustment (addition) to
+the value based on dfdl:prefixIncludesPrefixLength.
+
+The evaluation order of statements associated with prefix lengths is:
+
+1. prefix-prefix statement annotations
+
+2. prefix statement annotations
+
+3. statement annotations for the element itself per the description in
+Section 9.5 above.
+
+The statement annotations for a prefix (or prefix-prefix) are
+evaluated in a similar manner as if they were elements.
+The prefix (or prefix-prefix) length fields can be considered
+_quasi-elements_, keeping in mind that as they do not have element
+declarations these can never be nilled, nor can they have array
+dimension, be optional, be empty or have default values as those
+require XSD attributes or DFDL properties that are expressed only on
+element declarations/references.
+
+This implies that in terms of the rules in section 9.3.2 about
+establishing representation, prefix (or prefix-prefix) quasi-elements
+never have zero length, and always have simple normal representation.
+
+The evaluation order for the prefix (or prefix-prefix) statement annotations is:
+
+1. dfdl:discriminator or dfdl:assert(s) with testKind 'pattern' (parsing only)
+
+2. The prefix (or prefix-prefix) integer itself following property
+scoping rules, which includes establishing simple normal
+representation as described in Section 9.3.2 followed by conversion to
+a value of the simple integer type.
+
+3. dfdl:setVariable(s) - in lexical order, innermost schema component first
+
+4. dfdl:discriminator or dfdl:assert(s) with testKind 'expression'
+(parsing only)
+
```

Comment:
I'm not 100% convinced of this. I think that if I had a discriminator/assert with testKind 'pattern' on my element, I would want that evaluated **before** the prefix length was processed. It's peeking at the bit stream.

Reply:
I think there are two plausible fixes to make this clearer:

1. Allow only one of the prefix-prefix type, the prefixlengthType, or the element itself to have a pattern assert/discrim. 
2. Allow each to have one, and it is evaluated at the start of the corresponding grammar region

I'm going to suggest (2) is more natural, because we know we want to allow variables to be set based on the values of the prefix-prefix, and prefix quasi-elements, hence, there can be multiple dfdl:setVariable statements associated with each length type.  

So the different setVariable statements are not combined together across the prefix-prefix, prefix, and element itself; hence, it would be awkward to say that the pattern assert/discrim are so combined, but the setVariable statements are not.

Reply:
See the comment on the issue ticket here:

https://github.com/OpenGridForum/DFDL/issues/29#issuecomment-1516996502

For the example. We can discuss on a dfdl-wg call to review whether this is the proper evaluation order.